app/models/attribute.rb

```diff
@@ -4,4 +4,6 @@ class Attribute < ActiveRecord::Base
   belongs_to :resource
 
   has_many :translated_attributes
+
+  validates :key, format: { with: /\A[[:alpha:]]+(_[[:alpha:]]+)*\z/ }
```

Comment:
Yup that's it.  Might be too strict, but I can adjust it if needed.  Rails makes it easy 😄